### PR TITLE
CLOUDP-115725: AtlasCLI v1.0.0 show the new release alert message

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-115725](https://jira.mongodb.org/browse/CLOUDP-115725)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
➜  mongocli git:(CLOUDP-115725) mongoclidev config list
PROFILE NAME
default
test
➜  mongocli git:(CLOUDP-115725) make build-atlascli
==> Building atlas binary
go build -ldflags "-s -w -X github.com/mongodb/mongocli/internal/version.GitCommit=59dc12eb62551fb964cb7e7e2676ada5fd2e31c2 -X github.com/mongodb/mongocli/internal/config.ToolName=atlascli -X github.com/mongodb/mongocli/internal/version.Version=1.0.0-next" -o ./bin/atlas ./cmd/atlas
➜  mongocli git:(CLOUDP-115725) ./bin/atlas config list
PROFILE NAME
default
test
```

### CODE REVIEW

File: internal/latestrelease/finder.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func (f *finder) Find() (releaseInfo *ReleaseInformation, err error) {
 	}
 
 	releaseInfo, newestVersion, err := f.loadOrGet()
-	if err != nil || releaseInfo == nil || svCurrentVersion.Compare(newestVersion) > 0 {
+	if err != nil || releaseInfo == nil || svCurrentVersion.Compare(newestVersion) >= 0 {
```
Comment (andreaangiolillo) [inline internal/latestrelease/finder.go:129]: Could you check why we don't get this error with mongocli v1.23.1?

```
/Downloads/mongocli_1.23.1_macos_x86_64/bin ./mongocli iam projects list
ID                         NAME
5e4e593f70dfbf1010295836   CLI ANDREA
61dc598fd1f07c7b37ba425d   CLI Bianca
5e429f2e06822c6eac4d59c9   CLI tests
607caaa2b4c1fd299fb542be   Migration
613625603e005358482462d9   ProjectFoobar
61279659d7d1a561be8ab56e   TF GUS
```

Comment (andreaangiolillo) [reply internal/latestrelease/finder.go:129]: I have downloaded the 1.23.1 binary from here https://github.com/mongodb/mongodb-atlas-cli/releases/tag/mongocli%2Fv1.23.1

Comment (andreaangiolillo) [reply internal/latestrelease/finder.go:129]: you are running against master. This issue may have been introduced with one of your latest changes that are not in the 1.23.1

Comment (blva) [reply internal/latestrelease/finder.go:129]: Thanks, precisely. atlascli has my change whereas mcli didn't have. The bug was introduced with https://github.com/mongodb/mongodb-atlas-cli/pull/1010/files in the comparison. 

It was changed when I updated the logic which. was 
```
	if currentVersion.Compare(v) < 0 {
		return true, &version.ReleaseInformation{
			Version:     v.Original(),
			PublishedAt: release.GetPublishedAt().Time,
		}, nil
	}
``
`

**Comment by andreaangiolillo** [comment]
I see that in your example you are building atlascli with version `v1.0.0-next`. Could you build atlascli with the version `v1.0.0` and check that you don't get the message, please?

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**